### PR TITLE
Add amcl demo

### PR DESCRIPTION
**Pull request by dhood**
Requires ros2/turtlebot2_demo#38 and connected PRs

This adds a launch file to bring up the appropriate nodes for the ROS 2 AMCL demo on the turtlebot2.

Main question that I am looking for help with (we don't seem to have many examples of using `launch`):

How can I pass arguments through to the launch file? I would like to pass the map path through, [so I added an argument parser](https://github.com/ros2/turtlebot2_demo/compare/add_amcl_demo?expand=1#diff-b43222f51127654b8934a03ba8b9a6d6R25). However, `launch` is catching the arguments before they can get through to the launchfile.
```
ubuntu@pine64:~/tb2_demo_ws_dhood$ launch install_isolated/turtlebot2_localization/share/turtlebot2_localization/launch/turtlebot_amcl.py --map ~/maps/osrf_map.yaml
usage: launch [-h] launch_file [launch_file ...]
launch: error: unrecognized arguments: --map /home/ubuntu/maps/osrf_map.yaml
```

At the moment I'm depending on turtlebot2_drivers and have copy-pasted the "common" turtlebot nodes in the launch file. These will be replaced by a call to launch the common nodes when the launch file exists (there's some refactoring planned from what I understand).

### CODE REVIEW

**Comment by dhood** [comment]
@dirk-thomas suggested just not using the `launch` invocation at the moment: I will refactor `turtlebot_amcl.py`

**Comment by nuclearsandwich** [comment]
Is this demo targeting Beta 2?

**Comment by dhood** [comment]
yes please. there are a few new leaf packages that will be added with refactoring of this repo if I understood @mikaelarguedas' plan correctly

**Comment by mikaelarguedas** [comment]
> Is this demo targeting Beta 2?

Yep

> there are a few new leaf packages that will be added with refactoring of this repo if I understood @mikaelarguedas' plan correctly

Yep, the refactor is more of a nice to have, it should take very little time to rebuild and prevent turtlebot_drivers to depend on everything else because all the demos launchfiles are stored in this package

**Comment by clalancette** [comment]
> Yep, the refactor is more of a nice to have, it should take very little time to rebuild and prevent turtlebot_drivers to depend on everything else because all the demos launchfiles are stored in this package

@mikaelarguedas If you are too busy with the memory leaks and such, do you want me to do the turtlebot2_demo repository refactor?  I should have time today or Monday.

**Comment by mikaelarguedas** [comment]
Oh thanks for offering. I think this should be done before Monday so that everything is packaged for testing. If you happen to have time before tonight that's great, otherwise I'll do it either tonight or tomorrow. It should not take long to do it's only shuffling files around and update dependencies

**Comment by clalancette** [comment]
What I'm doing right at the moment isn't critical.  I'll get started right away.

**Comment by mikaelarguedas** [comment]
sounds good thanks!

**Comment by dhood** [comment]
This now requires ros2/turtlebot2_demo#38 and related PRs

File: turtlebot2_amcl/launch/turtlebot_amcl.py
Comment (dhood) [inline turtlebot2_amcl/launch/turtlebot_amcl.py:113]: amcl gets installed to the path at the moment because of https://github.com/ament/ament_cmake/issues/96

**Review by clalancette — APPROVED** [review]
This looks good to me.

File: turtlebot2_amcl/launch/turtlebot_amcl.py
Comment (mikaelarguedas) [reply turtlebot2_amcl/launch/turtlebot_amcl.py:113]: sounds good thanks for clarifying

**Review by mikaelarguedas — APPROVED** [review]
lgtm,
note for future iterations:
- The argument parser is placed within the launch function rather than main because launch should be able to handle arguments in the future
- amcl is on the path because of the default behavior of ament_cmake_auto

**Comment by mikaelarguedas** [comment]
I leave it to @dhood discretion to run CI or not on this before merging

**Comment by dhood** [comment]
ci_turtlebot-demo_linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_turtlebot-demo_linux-aarch64/81/)
ci_turtlebot-demo_linux [Build Status](http://ci.ros2.org/job/ci_turtlebot-demo_linux/66/)

**Pull request merged**
